scripts/check_golang_version.sh

```diff
@@ -16,8 +16,9 @@ set -eo pipefail
 # fail the first check (GO_MOD_SUPPORT), and this isn't necessary for travis builds.
 #
 # https://docs.travis-ci.com/user/environment-variables/#default-environment-variables
-if $CI && $TRAVIS
-then
+#
+# TRAVIS_BRANCH is set for external ARM builds, and CI/TRAVIS are not set
+if [[ -n $CI ]] && [[ -n $TRAVIS ]] || [[ -n $TRAVIS_BRANCH ]]; then
```

Comment:
generally speaking, you should avoid using travis-specific variables outside the "/scripts/travis" directory.

Reply:
It looks like a comment for #1702. Open to suggestions? call `./scripts/travis/istravis_build.sh` here?

Reply:
no, just export the path from the makefile. The reason it doesn't find the go compiler is because the makefile doesn't export the PATH to this script.

Reply:
It fails not from Makefile but from `travis/build.sh` when go is not installed:
```
if ! ./scripts/check_golang_version.sh
then
    exit 1
fi
# Get the go build version.
GOLANG_VERSION=$(./scripts/get_golang_version.sh)

curl -sL -o ~/gimme https://raw.githubusercontent.com/travis-ci/gimme/master/gimme
```

Reply:
just because the old condition `if $CI && $TRAVIS` returns true if CI/TRAVIS unset, as it happens in external build.

Reply:
so I guess that's where you need to fix it, right ? i.e. avoid checking the Golang version if no go is installed.

Reply:
I'm not sure what is better. Under CI you have either older go and `[ -n $CI ]] && [[ -n $TRAVIS ]]` skip that or no go at all. So the question why do check go version in `travis/build.sh` @btoll ?

Reply:
yes, agree with @algorandskiy ; sounds to me like this check isn't very meaningful in travis specific executions since it's getting skipped anyway.